homeassistant/components/daikin/__init__.py

```diff
@@ -127,3 +130,71 @@ def device_info(self) -> DeviceInfo:
             name=info.get("name"),
             sw_version=info.get("ver", "").replace("_", "."),
         )
+
+
+async def async_migrate_unique_id(
+    hass: HomeAssistant, config_entry: ConfigEntry, api: DaikinApi
+) -> None:
+    """Migrate old entry."""
+    dev_reg = dr.async_get(hass)
+    old_unique_id = config_entry.unique_id
+    new_unique_id = api.device.mac
+    new_name = api.device.values.get("name")
+
+    @callback
+    def _update_unique_id(entity_entry: er.RegistryEntry) -> dict[str, str] | None:
+        """Update unique ID of entity entry."""
+        return update_unique_id(entity_entry, new_unique_id)
+
+    if new_unique_id != old_unique_id:
+        # Migrate devices
+        for device_entry in dr.async_entries_for_config_entry(
+            dev_reg, config_entry.entry_id
+        ):
+            for connection in device_entry.connections:
+                if connection[1] == old_unique_id:
+                    new_connections = {
+                        (CONNECTION_NETWORK_MAC, dr.format_mac(new_unique_id))
+                    }
+                    _LOGGER.info(
+                        "Migrating device %s to %s and connections to %s",
+                        device_entry.name,
+                        new_name,
+                        new_connections,
+                    )
+                    dev_reg.async_update_device(
+                        device_entry.id,
+                        name=new_name,
+                        merge_connections=new_connections,
+                    )
+
+        # Migrate entities
+        await er.async_migrate_entries(hass, config_entry.entry_id, _update_unique_id)
+
+        new_data = {**config_entry.data, KEY_MAC: dr.format_mac(new_unique_id)}
+
+        hass.config_entries.async_update_entry(
+            config_entry, unique_id=new_unique_id, data=new_data
+        )
+
+
+@callback
+def update_unique_id(
+    entity_entry: er.RegistryEntry, unique_id: str
+) -> dict[str, str] | None:
+    """Update unique ID of entity entry."""
+    if entity_entry.unique_id.startswith(unique_id):
+        # Already correct, nothing to do
+        return None
+
+    unique_id_parts = entity_entry.unique_id.split("-")
+    unique_id_parts[0] = unique_id
+    entity_new_unique_id = "-".join(unique_id_parts)
+
+    _LOGGER.info(
```

Comment:
Max debug log please.
```suggestion
    _LOGGER.debug(
```